### PR TITLE
nginx: Uninstall pf and install launch_socket_server

### PR DESCRIPTION
**Pull request by mistydemeo**
cc @MikeMcQuaid 

Refs github/friction#19.

### CODE REVIEW

File: cmd/brew-setup-nginx-conf [hunk 1]
```diff
@@ -79,6 +79,15 @@ if `readlink /etc/resolver 2>/dev/null`.chomp != "/usr/local/etc/resolver"
   end
 end
 
+if File.exist? "/etc/pf.anchors/dev.strap"
+  unless system "sudo -n true >/dev/null"
+    puts "Asking for your password to setup launch_socket_server:"
+  end
+  system "sudo rm /etc/pf.anchors/dev.strap"
+  system "brew install launch_socket_server"
+  system "brew services restart launch_socket_server"
```
Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: Need to use `sudo` for this.

File: cmd/brew-setup-nginx-conf [hunk 2]
```diff
@@ -79,6 +79,15 @@ if `readlink /etc/resolver 2>/dev/null`.chomp != "/usr/local/etc/resolver"
   end
 end
 
+if File.exist? "/etc/pf.anchors/dev.strap"
+  unless system "sudo -n true >/dev/null"
+    puts "Asking for your password to setup launch_socket_server:"
+  end
+  system "sudo rm /etc/pf.anchors/dev.strap"
```
Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: Also want to delete `/etc/pf.anchors/dev.github` and remove `dev.strap` references from `/etc/pf.conf` e.g. with `grep -v "dev.strap" /etc/pf.conf` written to a temporary file.

File: cmd/brew-setup-nginx-conf (hunk 1 above)
Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: (as it's a system-level service)

File: cmd/brew-setup-nginx-conf [hunk 3]
```diff
@@ -79,6 +79,15 @@ if `readlink /etc/resolver 2>/dev/null`.chomp != "/usr/local/etc/resolver"
...
+  system "brew install launch_socket_server"
```
Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: Could just add this to `brewfile` on line 58? Trying to somewhat deprecate manual `brew install` calls in our scripts.

File: cmd/brew-setup-nginx-conf (hunk 1 above)
Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: Also probably want to rework `started_nginx` below to be `brewfile_installs` or something and then restart both `nginx` and `launch_socket_server` and check `launch_socket_server`'s status with e.g. `unless system "sudo brew services restart launch_socket_server >/dev/null"`. Maybe also worth adding a `sudo -k` at the end of this file, too (which I'd forgotten to do previously).

**Comment by MikeMcQuaid** [comment]
@mistydemeo Thanks for jumping in on this! So good to not have to work on this stuff alone!

File: cmd/brew-setup-nginx-conf (hunk 2 above)
Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: And probably `launchctl unload /Library/LaunchDaemons/dev.strap.pf.plist` and remove it too.

Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: Actually probably want to modify those files, unload it, load it again, unload it again and delete the `plist` 😅

Comment (MikeMcQuaid) [inline cmd/brew-setup-nginx-conf]: Have addressed this, all fine now 👍

**Comment by MikeMcQuaid** [comment]
@mistydemeo I've made changes for my comments. Let me know if you have any thoughts on them.

**Comment by mistydemeo** [comment]
:thumbsup:

**Pull request merged**